### ISSUE TITLE
Pinia nuxt version 0.6.0+ causes error when used with pnpm monorepo

### ISSUE THREAD

**Issue by kalvenschraut** [posva/pinia#2820]
### Reproduction

https://stackblitz.com/~/github.com/kalvenschraut/pinia-nuxt-pnpm-monorepo-issue

### Steps to reproduce the bug

To see issue do
```shell
pnpm install
pnpm --filter website1 run dev
```

can then downgrade @pinia/nuxt to 0.5.5 and run same commands to see it working. Can also upgrade to 0.6.1 to verify still not working
```shell
pnpm update @pinia/nuxt@0.5.5 -r
pnpm --filter website1 run dev
```

### Expected behavior

Page to load and show the users name from the user store

### Actual behavior

![image](https://github.com/user-attachments/assets/6c7edb2d-61c1-4d60-bd86-939bd24aabf5)

### Additional information

In prod builds on my personal site where I first initial saw this was seeing vueDemi.effectScope is not a function error. Downgrading back to 0.5.5 for now until direction can be given

**Comment by rylanharper** [comment]
Can confirm. This is breaking all my Nuxt projects that use Pinia 😢

**Comment by maxdzin** [comment]
Same for my Nuxt + Pinia projects

**Comment by posva** [comment]
@rylanharper @maxdzin just thumbs up the issue, that way mantainers can sort issues and find the most searched ones 😉

**Comment by posva** [comment]
I did some research, and the problem persists even with upgraded Nuxt and pnpm. The cause of the problem is the removal of the alias for pinia in https://github.com/vuejs/pinia/commit/65031ee77ed46a34bc2359223e24c7944e840819
I needed that change to work in other scenarios so I will investigate this further. In the meantime, here is a workaround: add an alias to your nuxt.config so pinia is resolved correctly:

```ts
import { fileURLToPath } from 'node:url'
import path from 'node:path'

const rootDir = fileURLToPath(new URL('../..', import.meta.url))

export default defineNuxtConfig({
  modules: [ '@pinia/nuxt'],
  alias: {
   pinia: path.resolve(rootDir, 'node_modules/pinia/dist/pinia.mjs')
  }
});
```

Or exclude pinia from vite optimization:

```ts
export default defineNuxtConfig({
  modules: [ '@pinia/nuxt'],
  vite: {
    optimizeDeps: {
      exclude: ['pinia']
    }
  },
});
```

@danielroe I would like some help with this matter: in my tests, I have two copies of pinia in non monorepo nuxt applications. I still don't understand why this part of the Pinia module is so flaky, and I suppose it's more of an issue of Vite than Nuxt. But I imagine you have seen this issue often enough to guide me in the right direction (and add a note comment so my future self does not forget 😅)

**Comment by posva** [comment]
Using the `shamefully-hoist=false` also works

**Comment by posva** [comment]
I'm still not 100% sure of the fix being the right solution so I will keep this open until then

**Comment by HP8585** [comment]
please solve the problem I keep getting:
500
[vite-node] [ERR_LOAD_URL] pinia

at pinia

**Comment by posva** [comment]
@HP8585 thats an npm issue #2822

**Comment by jaap3** [comment]
I've updated @nuxt/pinia to 0.7.0 last week and started running into this issue. Downgrading to 0.6.1 makes everything work again.

Additional data points:

* I'm also on a monorepo, but using plain npm
* nuxt: ~3.14.159
* pinia: ~2.2.6
* Using nuxt layers, pinia and @nuxt/pinia are installed in the "bottom" layer

EDIT: So 0.6.1 works in development (at least for me), but after a deploy to staging "everything" was broken. Only after downgrading to 0.5.5 did a deployed build work again.

**Comment by ibrahimzdmr** [comment]
Hello, today I've updated from v0.5.5 to v0.7.0 and couldn't run the project.

This is the error I'm receiving "[nuxt] [request error] [unhandled] [500] Cannot find module '/.output/server/node_modules/pinia/dist/pinia.mjs' imported from /.output/server/chunks/build/server.mjs"

and when I check the file in node module there is only pinia.prod.cjs file.

nuxt: 3.13.2

also I use pnpm

Thank you

**Comment by posva** [comment]
I think that to fix this we might release a new version that removes the compatibility with Vue 2.6 but keeps Vue 2.7. This will allow removing some peer dependencies that nom isn't able to support (like other package managers do)

**Comment by HP8585** [comment]
please just fix it whatever changes it takes let it take...

**Comment by Moonrag** [comment]
Same issue here with pnpm on 0.6.0+ and prod build

just checked with blank project
node v20.18.1
pnpm 9.14.2

reproduction:
```bash
pnpx nuxi@latest init nuxt-test
pnpx nuxi@latest module add pinia

pnpm build && pnpm preview
```

<img width="1624" alt="image" src="https://github.com/user-attachments/assets/cc41c0b9-d6a6-47b0-a781-1299601beec3">

**Comment by frontwero** [comment]
Same issue here as @Moonrag with node v20.17.0 and npm 10.8.2
When I upgrade from @pinia/nuxt 0.5.5 to 0.6.0+ I get the same error "vueDemi2.effectScope is not a function".
Downgrade to 0.5.5 and it works again.
Using nuxt 3.14.159 and pinia 2.2.6

**Comment by posva** [comment]
That error is fixed by installing pinia

**Comment by posva** [comment]
This was partially fixed en #2846 but requires https://github.com/unjs/nypm/pull/165 to be merged and released so adding the Nuxt module **also adds `pinia`** as a dependency to your Nuxt project. In the meantime, or if you already have the `@pinia/nuxt` module installed, install Pinia alongside the Nuxt module. I will update docs.

**Comment by pi0** [comment]
https://github.com/unjs/nypm/releases/tag/v0.4.1

**Comment by danielroe** [comment]
this should now be resolved in latest nuxi! 🎉

**Comment by slavanossar** [comment]
Can [this issue](https://github.com/vuejs/pinia/issues/2837) be reopened? It still persists after upgrading to `@pinia/nuxt@0.8.0` and regenerating a fresh lockfile, I'm not entirely sure it's the same issue as above.

I'm also using a `pnpm` monorepo, and I already had `pinia` as a dependency. The only way to resolve was downgrading to `@pinia/nuxt@0.5.5`.

Thanks!

**Comment by posva** [comment]
If you have a boiled down without Apollo or other libraries, I can take a look

**Comment by slavanossar** [comment]
I'll try to put together a repro for you over the weekend, much appreciated!

**Comment by Felixclement** [comment]
im my case this issue was resolved from using a more recent node image slim-23

**Comment by martin-djukanovic** [comment]
> Can [this issue](https://github.com/vuejs/pinia/issues/2837) be reopened? It still persists after upgrading to `@pinia/nuxt@0.8.0` and regenerating a fresh lockfile, I'm not entirely sure it's the same issue as above.
> 
> I'm also using a `pnpm` monorepo, and I already had `pinia` as a dependency. The only way to resolve was downgrading to `@pinia/nuxt@0.5.5`.
> 
> Thanks!

Maybe try the following? In `node_modules/pinia/dist/pinia.mjs` change the definition of `shouldHydrate(` on line ~1211 by replacing `obj.hasOwnProperty(skipHydrateSymbol)` with `!Object.prototype.hasOwnProperty.call(obj, skipHydrateSymbol)`. The payload-plugin is the reason why it works on 0.5.5 but not 0.6.0.

I tried to add `console.log()` to `shouldHydrate(` to verify that the apollo response is the object that causes the issue, but that caused the server to hang for some reason. :confused:

**Comment by leon-wbr** [comment]
> Maybe try the following? In `node_modules/pinia/dist/pinia.mjs` change the definition of `shouldHydrate(` on line ~1211 by replacing `obj.hasOwnProperty(skipHydrateSymbol)` with `!Object.prototype.hasOwnProperty.call(obj, skipHydrateSymbol)`. The payload-plugin is the reason why it works on 0.5.5 but not 0.6.0.

I also had to do this using a yarn monorepo. I'm not sure what is causing it, but it's definitely a bug.

**Comment by mobayen** [comment]
Hi all,  

I am still encountering this issue, which is identical to #2837.  

The problem occurs after upgrading `@pinia/nuxt` from `0.5.1` to `0.9.0`. It specifically happens when making GraphQL requests using `useAsyncQuery (Apollo client)` or `useQuery()`.  

Any help or insights would be greatly appreciated!  
BTW, i tried everything i found so far!

**Comment by kalvenschraut** [comment]
For my original issue it is resolved me for, I just had to add pinia as a direct dependency of my nuxt apps.

**Comment by kyurkchyan** [comment]
@mobayen  I have been experiencing the same. I've created a new issue #2872 with reproducation repo.

**Comment by ejerskov** [comment]
Anyone who has a reliable workaround?

I'm trying to upgrade @nuxt/pinia from 0.5.5 to 0.9.0 but any upgrade to 0.6+ results in:

> "getActivePinia()" was called but there was no active Pinia.

Im running Nuxt 3 in a layered app, and Im installing in /app.

I've tried some of the suggestions here:

- Adding alias to nuxt.config.ts but no matter what path I try to provide, the alias is "not found"
- Adding ['pinia'] to vite.optimizeDeps.exclude
- Install "pinia" along side "@nuxt/pinia", just "pnpm install pinia"

But none of these work - anyone with other suggestions?

**Comment by DiyanDev** [comment]
The issue persists in @nuxt/pinia version 0.10.0 with Nuxt 3.15.4, despite having Pinia as a direct dependency.

I was able to resolve it by applying the alias workaround suggested by @posva .

In my case, I'm using an internal component library that includes its own store. Within the library, I marked Pinia as an external dependency in the Vite config as follows:

In the library I have marked pinia as an external dependency in vite config like so:
```            rollupOptions: {
                external: ["pinia"],
                output: {
                    exports: "named",
                    globals: {
                        pinia: "Pinia"
                    }
                }
            }
```

**Comment by Saimo** [comment]
**This could be a special case for me but I was able to fix it.**

I had
/main-nuxt
/layer-nuxt

main-nuxt used layer-nuxt.
In layer-nuxt I had one store in /stores
installed was nuxt-pinia and pinia (no idea why i installed pinia and the nuxt pinia but it was/is and it seems to work prod/dev)

 **package.json:**
"@pinia/nuxt": "0.10.1",
"pinia": "^3.0.1"
Installed in both nuxt apps

**How I fixed it:**
In the /stores stores of the layer-nuxt I was using 
`import { defineStore } from 'pinia'`

Removing this solved the issue completely.

**Comment by mrleblanc101** [comment]
@Saimo 

> no idea why i installed pinia and the nuxt pinia but it was/is and it seems to work prod/dev

Probably because that's what the instruction says:

> This will add both @pinia/nuxt and pinia to your project. If you notice that pinia is not installed, please install it manually with your package manager: npm i pinia.
